java/server/src/org/openqa/grid/internal/Registry.java

```diff
@@ -19,421 +19,240 @@
 
-import org.openqa.grid.internal.listeners.RegistrationListener;
-import org.openqa.grid.internal.listeners.SelfHealingProxy;
 import org.openqa.grid.internal.utils.configuration.GridHubConfiguration;
 import org.openqa.grid.web.Hub;
 import org.openqa.grid.web.servlet.handler.RequestHandler;
 import org.openqa.selenium.remote.DesiredCapabilities;
 import org.openqa.selenium.remote.internal.HttpClientFactory;
-import org.openqa.selenium.remote.server.log.LoggingManager;
 
 import java.util.List;
 import java.util.Set;
-import java.util.concurrent.CopyOnWriteArrayList;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.locks.Condition;
-import java.util.concurrent.locks.ReentrantLock;
-import java.util.logging.Level;
-import java.util.logging.Logger;
 
 /**
- * Kernel of the grid. Keeps track of what's happening, what's free/used and assigned resources to
+ * Deprecated Kernel of the grid. Keeps track of what's happening, what's free/used and assigns resources to
  * incoming requests.
+ *
+ * @deprecated use {@link DefaultGridRegistry} instead. This class is just a proxy to it.
  */
+@Deprecated
 @ThreadSafe
-public class Registry {
-
-  public static final String KEY = Registry.class.getName();
-  private static final Logger LOG = Logger.getLogger(Registry.class.getName());
-
-  // lock for anything modifying the tests session currently running on this
-  // registry.
-  private final ReentrantLock lock = new ReentrantLock();
-  private final Condition testSessionAvailable = lock.newCondition();
-  private final ProxySet proxies;
-  private final ActiveTestSessions activeTestSessions = new ActiveTestSessions();
-  private final GridHubConfiguration configuration;
-  private final HttpClientFactory httpClientFactory;
-  private final NewSessionRequestQueue newSessionQueue;
-  private final Matcher matcherThread = new Matcher();
-  private final List<RemoteProxy> registeringProxies = new CopyOnWriteArrayList<>();
-
-  private volatile boolean stop = false;
-  // The following three variables need to be volatile because we expose a public setters
-  private volatile Hub hub;
-
-  private Registry(Hub hub, GridHubConfiguration config) {
-    this.hub = hub;
-    this.newSessionQueue = new NewSessionRequestQueue();
-    this.configuration = config;
-    this.httpClientFactory = new HttpClientFactory();
-    proxies = new ProxySet(config.throwOnCapabilityNotPresent);
-    this.matcherThread.setUncaughtExceptionHandler(new UncaughtExceptionHandler());
+public class Registry implements GridRegistry {
+  BaseGridRegistry gridRegistry;
+
+  @Deprecated
+  public Registry() {
+    gridRegistry = new DefaultGridRegistry();
+  }
+
+  private Registry(Hub hub) {
+    gridRegistry = new DefaultGridRegistry(hub);
   }
 
+  private void setConfiguration(GridHubConfiguration configuration) {
+    gridRegistry.configuration = configuration;
+  }
 
+  /**
+   * @see GridRegistry#start()
+   */
+  @Deprecated
```

Comment:
why are you adding this as a deprecated method?

If it is already deprecated, can you add what the alternative should be?

Reply:
Hi @lukeis per the commit comments, my intention here is to deprecate the entire existing class `Registry` and replace it with `DefaultRegistry` -- which all code has be updated to use. The `Registry` class was left (and modified to use `DefaultRegistry` simply ease with compilation and notification, as people migrate, in case someone has custom grid code in their repository.

Reply:
@lukeis thanks for taking a look, btw!